spark/dl/src/main/scala/com/intel/analytics/bigdl/transform/vision/image/augmentation/RandomCropper.scala

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.transform.vision.image.augmentation
+
+import com.intel.analytics.bigdl.dataset.image.{CropCenter, CropRandom, CropperMethod}
+import com.intel.analytics.bigdl.transform.vision.image.{FeatureTransformer, ImageFeature}
+import com.intel.analytics.bigdl.transform.vision.image.opencv.OpenCVMat
+import com.intel.analytics.bigdl.utils.RandomGenerator
+import org.opencv.core.CvType
+
+object RandomCropper {
+  def apply(cropWidth: Int, cropHeight: Int,
+            mirror: Boolean, cropperMethod: CropperMethod = CropRandom): RandomCropper =
+    new RandomCropper(cropHeight, cropWidth, mirror, cropperMethod)
+}
+
+/**
+ * Random cropper on uniform distribution with fixed height & width
+ * @param cropWidth  width cropped to
+ * @param cropHeight height cropped to
+ * @param mirror   whether mirror
+ * @param cropperMethod crop method
+ */
+class RandomCropper(cropWidth: Int, cropHeight: Int,
+                      mirror: Boolean, cropperMethod: CropperMethod = CropRandom)
+  extends FeatureTransformer {
+
+  import com.intel.analytics.bigdl.utils.RandomGenerator.RNG
+
+  override protected def transformMat(feature: ImageFeature): Unit = {
+    val openCVMat = feature.opencvMat()
+
+    if (openCVMat.`type`() != CvType.CV_32FC3) {
+      openCVMat.convertTo(openCVMat, CvType.CV_32FC3)
+    }
+
+    val tmp = new Array[Float](openCVMat.height() * openCVMat.width() * openCVMat.channels())
```

Comment:
reuse the buffer?

Reply:
this is image specific, cannot make it global